lib/Importer/ONNX.cpp

```diff
@@ -502,6 +502,36 @@ std::unique_ptr<ONNXModelLoader> ONNXModelLoader::parse(const void *onnxModel,
 ONNXModelLoader::ONNXModelLoader(Function &F)
     : CommonOperatorLoader({}, {}, F) {}
 
+void ONNXModelLoader::checkInputs(onnx::GraphProto &net,
+                                  llvm::ArrayRef<const char *> names,
+                                  llvm::ArrayRef<Tensor *> tensors) {
+  for (size_t i = 0; i < names.size(); i++) {
+    // Look if a corresponding input exists.
+    for (int j = 0; j < net.input_size(); j++) {
+      const onnx::ValueInfoProto &valueInfo = net.input(j);
+      const std::string &inputName = valueInfo.name();
+
+      if (inputName == names[i]) {
+        llvm::ArrayRef<size_t> dims = tensors[i]->dims();
+        const onnx::TensorShapeProto &shape =
+            valueInfo.type().tensor_type().shape();
+
+        // Check if the number of dimensions is consistent.
+        if (dims.size() != shape.dim_size()) {
+          llvm_unreachable("Mismatch between input image and ONNX input shape");
+        }
+        // Allow batch dimensions to be different.
+        for (size_t k = 1; k < dims.size(); k++) {
+          if (dims[k] != shape.dim(k).dim_value()) {
+            llvm_unreachable(
+                "Mismatch between input image and ONNX input shape");
+          }
```

Comment:
This code can be simplified. Instead of increasing the nest level please use early exits. 

https://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code

Reply:
Removed one nesting level (replaced by assert). I assume this is what you meant.